newsroom/oauth_clients/clients_async.py

```diff
@@ -0,0 +1,47 @@
+from content_api import MONGO_PREFIX
+from typing import Optional, Annotated, List, Dict, Union
+from superdesk.core.resources import ResourceModel, ResourceConfig, MongoResourceConfig
+from superdesk.core.resources.service import AsyncResourceService
+from superdesk.core.web import EndpointGroup
+from pydantic import Field
+import logging
+from bson import ObjectId
+
+
+class ClientResource(ResourceModel):
+    id: Annotated[Union[str, ObjectId], Field(alias="_id")] = None
+    name: str
+    password: str
+    last_active: Optional[str] = None
+    etag: Annotated[Optional[str], Field(alias="_etag")] = None
+
+
+class ClientService(AsyncResourceService[ClientResource]):
+    """Service class for managing OAuthClient resources"""
+
+    resource_name = "oauth_clients"
+
+    async def get_all_client(self) -> List[Dict]:
```

Comment:
The naming of this function should be plural, as it potentially returns more than 1